include/mbedtls/cipher.h

```diff
@@ -293,21 +293,20 @@ typedef struct mbedtls_cipher_info_t {
      * For ciphers that accept variable IV sizes,
      * this is the recommended size.
      */
-    unsigned int MBEDTLS_PRIVATE(iv_size);
+    uint16_t MBEDTLS_PRIVATE(iv_size);
 
     /** Bitflag comprised of MBEDTLS_CIPHER_VARIABLE_IV_LEN and
      *  MBEDTLS_CIPHER_VARIABLE_KEY_LEN indicating whether the
      *  cipher supports variable IV or variable key sizes, respectively.
      */
-    int MBEDTLS_PRIVATE(flags);
+    uint8_t MBEDTLS_PRIVATE(flags);
 
     /** The block size, in Bytes. */
-    unsigned int MBEDTLS_PRIVATE(block_size);
+    uint8_t MBEDTLS_PRIVATE(block_size);
 
     /** Struct for base cipher information and functions. */
     const mbedtls_cipher_base_t *MBEDTLS_PRIVATE(base);
-
-} mbedtls_cipher_info_t;
+} MBEDTLS_PACKED mbedtls_cipher_info_t;
```

Comment:
In general, you can't put packed structures in the public API. The library and the code that uses it may be compiled with different compilers, so we can't use non-portable things that change the ABI.

In this case, we _might_ get away with it if we make sure there are no `static inline` functions, because we know all the `mbedtls_cipher_info_t` structures are the static ones in `cipher_wrap.c`. But I'd much rather not play with fire, and instead reorder fields to be naturally packed.